cobigen-cli/cli/src/main/java/com/devonfw/cobigen/cli/commands/GenerateCommand.java

```diff
@@ -140,14 +138,33 @@ public Integer doAction() throws Exception {
     boolean firstIteration = true;
     List<T> finalTos = new ArrayList<>();
     List<Object> generationInputs = new ArrayList<>();
-    for (File inputFile : this.inputFiles) {
+    for (Path inputFile : this.inputFiles) {
 
-      String extension = inputFile.getName().toLowerCase();
+      String extension = inputFile.getFileName().toString().toLowerCase();
       boolean isJavaInput = extension.endsWith(".java");
       boolean isOpenApiInput = extension.endsWith(".yaml") || extension.endsWith(".yml");
 
+      // checks for output root path and project root being detectable
+      if (this.outputRootPath == null && MavenUtil.getProjectRoot(inputFile, false) == null) {
+
+        LOG.info(
+            "Did not detect the input as part of a maven project, the root directory of the maven project was not found.");
+
+        // cancels with an error if more than one input file was provided
+        if (this.inputFiles.size() > 1) {
+          LOG.error(
+              "As multiple input files were provided, setting an automatic output path is not possible. Cancelling...");
+          System.exit(1);
+        }
```

Comment:
I think we can remove this abortion for multiple files if we would simply propose to use pwd as the root output path

Reply:
Not sure about that, because if the user provides several input files, we can't ensure that he wants all files to be generated to the same directory.

Reply:
let's assume that, yes
Let's continue then, otherwise the user will be blocked all the time and he/she in any case has to provide one output path also by parameters. There is also no way of differentiation.

Reply:
Ok, I can agree to that. Removed now.